fix: token space race condition

r4topunk opened the pull request:
This pull request introduces several enhancements and fixes across multiple files to improve parameter handling, support contract-based spaces, and streamline API functionality. The changes primarily focus on resolving parameters asynchronously, adding support for contract metadata, and refining API endpoints for better error handling and data retrieval.

### Enhancements to parameter handling:

* `src/app/(spaces)/p/[proposalId]/[tabname]/page.tsx`: Updated to resolve `params` asynchronously before extracting `proposalId`, ensuring proper handling of dynamic parameters. ([src/app/(spaces)/p/[proposalId]/[tabname]/page.tsxL11-R12](diffhunk://#diff-71fbe1e318bcec44c4a347cee060013c085c9d19a3d6e48deff6e0f5a7b582a4L11-R12))
* `src/app/(spaces)/t/[network]/[contractAddress]/page.tsx`: Modified to resolve `params` asynchronously before extracting `network` and `contractAddress`, improving consistency in parameter handling. ([src/app/(spaces)/t/[network]/[contractAddress]/page.tsxR22](diffhunk://#diff-8ec0274da43bfd2db101ad033fa31660263b5583701d6c172fa98fa7a5a9526dR22), [src/app/(spaces)/t/[network]/[contractAddress]/page.tsxL31-R33](diffhunk://#diff-8ec0274da43bfd2db101ad033fa31660263b5583701d6c172fa98fa7a5a9526dL31-R33))
* `src/app/explore/[slug]/page.tsx`: Updated to resolve `params` asynchronously before retrieving the `slug`, ensuring compatibility with dynamic routing. ([src/app/explore/[slug]/page.tsxL59-R60](diffhunk://#diff-90e2eb8852f8cdce738bde0a52f253be15a3ed73f0ced6dbc28d64c4e2d69a90L59-R60))

### Support for contract-based spaces:

* [`src/common/data/stores/app/space/spaceStore.ts`](diffhunk://#diff-98fd495414735652a5f73447d8c21927977f2526811c440d6ced426a573a9a3bR1079-R1098): Enhanced the `createSpaceStoreFunc` to populate `localSpaces` with metadata for contract spaces, including `contractAddress` and `network`. This ensures accurate representation of contract-based spaces in the store.
* [`src/pages/api/space/registry/from-contract.ts`](diffhunk://#diff-b9cd7b0c1e89b6cc4e817ed1c14a491ae93cb57e0b9673cbd0212c27176ee185R1-R43): Added a new API endpoint to fetch the `spaceId` associated with a specific `contractAddress` and `network`. This improves integration with contract-based spaces.

### API refinements:

* `src/pages/api/space/registry/[spaceId]/index.ts`: Extended the `identitiesCanModifySpace` function to include `network` in the query and handle cases where the network is missing, improving error handling and data accuracy. ([src/pages/api/space/registry/[spaceId]/index.tsL140-R158](diffhunk://#diff-841f713350f1aa369b123b5d49e30cff9d3a5f7cb8071917bf9895313cff7064L140-R158))
* `src/pages/api/space/registry/[spaceId]/tabs/[tabId].ts`: Refactored the `updateSpace` function to avoid direct mutation of the `req` object, improving code clarity and ensuring proper handling of the `network` parameter. ([src/pages/api/space/registry/[spaceId]/tabs/[tabId].tsL64-R68](diffhunk://#diff-a9de1aaaf8ce5ea9e7dcf062e33cc5806b65994920cc4dc897ceb80e38730238L64-R68), [src/pages/api/space/registry/[spaceId]/tabs/[tabId].tsL76-R78](diffhunk://#diff-a9de1aaaf8ce5ea9e7dcf062e33cc5806b65994920cc4dc897ceb80e38730238L76-R78), [src/pages/api/space/registry/[spaceId]/tabs/[tabId].tsL114-R116](diffhunk://#diff-a9de1aaaf8ce5ea9e7dcf062e33cc5806b65994920cc4dc897ceb80e38730238L114-R116))

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Improved contract space lookup and loading, including support for fetching space IDs by contract address and network.
  * Enhanced validation and error handling for invalid or missing route parameters.
  * More granular and informative error responses in API endpoints.

* **Bug Fixes**
  * Ensured contract and network metadata are consistently updated and stored for editable spaces.
  * Prevented unintended mutation of request objects when updating space data.

* **Refactor**
  * Streamlined internal logic for asynchronous parameter resolution and API request handling for better reliability and maintainability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

r4topunk committed:
♻️ refactor: resolve params in WrapperProposalPrimarySpace and related components

Updated the parameter handling in multiple components to ensure
params are resolved before usage. This improves code clarity and
prevents potential issues with undefined values.

r4topunk committed:
✨ feat: add API endpoint to fetch space by contract address and network

- Implemented a new API route to retrieve space information based on
  contract address and network.
- Enhanced existing functions to handle network data and improve error
  handling for better user feedback.

r4topunk committed:
✨ feat: add loading spinner for space fetching in ContractPrimarySpaceContent

- Introduced a loading state to display a spinner while fetching the
  space ID for the contract. This improves user experience by providing
  visual feedback during data retrieval.

r4topunk committed:
♻️ refactor: remove unused tokenData prop from ContractPrimarySpaceContent

The tokenData prop was removed as it was not being utilized within the
component, streamlining the code and improving readability.

r4topunk committed:
🎨 style: center loading spinner in ContractPrimarySpaceContent

Updated the loading spinner to be centered within a full-screen div
for improved user experience during space fetching.

sktbrd committed:
♻️ Adressing CodeRabbit: update spaceId handling in ContractPrimarySpaceContent and improve slug validation in Post component

j-paterson commented on src/pages/api/space/registry/[spaceId]/tabs/[tabId].ts:
```diff
@@ -61,9 +61,11 @@ async function updateSpace(
 
-  if (!isSignedFile(req)) {
+  const fileData = { ...req };
+  if (fileData.network) fileData.network = undefined;
```
Why are we setting network to undefined here?

sktbrd replied:
-  if (fileData.network) delete fileData.network;
+  if (fileData.network) fileData.network = undefined;


It was deleting already , we made it undefined

j-paterson commented on src/pages/api/space/registry/from-contract.ts:
Doesn't this duplicate functionality from the [spaceId] index registry system?

sktbrd replied:
duplicated api endpoint fixed in rato commit bellow

r4topunk committed:
✨ feat: consolidate space fetching logic into a single API endpoint

Replaced the separate 'from-contract' endpoint with a unified handler
that supports both contract lookups and listing modifiable spaces.
This change simplifies the API structure and improves maintainability.
